src/bin/hexyl.rs

```diff
@@ -179,23 +182,50 @@ fn main() {
     }
 }
 
-fn parse_byte_count(n: &str, block_size: u64) -> Option<u64> {
+#[derive(Clone, Copy, Debug, Default, Hash, Eq, Ord, PartialEq, PartialOrd)]
+pub struct PositiveI64(i64);
+
+impl PositiveI64 {
+    pub fn new(x: i64) -> Option<Self> {
+        if x.is_negative() {
+            None
+        } else {
+            Some(Self(x))
+        }
+    }
+
+    pub fn into_inner(self) -> i64 {
+        let Self(x) = self;
+        x
```

Comment:
This could also be written as `self.0`.

Reply:
It could. In general, I like to destructure (to make refactoring easier), but I DO prefer codebase consistency and didn't take the time to examine what other chunks of the codebase are doing, oops. I can change that here, since it's very unlikely things will change in the future.

Reply:
I've updated to use `self.0`.

Reply:
> In general, I like to destructure (to make refactoring easier),

I see. In this case I would probably just go with a regular struct instead of a tuple and then return `self.inner` or similar. But thank you for the explanation!

> but I DO prefer codebase consistency and didn't take the time to examine what other chunks of the codebase are doing, oops

To be honest: I'm not particular proud of `hexyl`s code base. I have never had to time to clean things up before (and after) contributions starting coming in. In this sense, there's not really one style of coding being used here. But thank you for considering this aspect!